polonius-engine/src/output/naive.rs

```diff
@@ -68,6 +76,20 @@ pub(super) fn compute<T: FactTypes>(
                 .map(|&(origin, point)| ((origin, point), ())),
         );
 
+        // Placeholder loans are contained by their placeholder origin at all points of the CFG.
+        //
+        // contains(Origin, Loan, Node) :-
+        //   cfg_node(Node),
+        //   placeholder(Origin, Loan).
```

Comment:
In the formal rules, we didn't have this, right? I'm fine with including it for now, though eventually (as we noted) we might want to try and refactor how we handle these "true everywhere" relations.

Reply:
by "in the formal rules", do you mean the [canonical rules hackmd](https://hackmd.io/CGMNjt1hR_qYtsR9hgdGmw) ? If so, we did, in order to not seed the placeholder loans at the root of the CFG — but the comment indeed refers to the slightly older predicate name of:

```prolog
origin_contains_loan_on_entry(Origin, Loan, Node) :-
  cfg_node(Node),
  placeholder(Origin, Loan).
```
which is around the end of the [borrowck paragraph](https://hackmd.io/CGMNjt1hR_qYtsR9hgdGmw#Subset-and-contains-the-heart-of-the-borrow-check).

Reply:
(and yeah it'd be great to refactor all the facts we materialize everywhere into a scalable and efficient solution)

Reply:
OK, and yeah I meant the hackmd.